IOW-556  Added tables: field_visit_readings_by_loc and datum_converte…

orosz-usgs opened the pull request:
…d_values

Before making a pull request
----------------------------

- [ ] Make sure all tests run
- [ ] Update the changelog appropriately

Title
-----------
Brief description of changes. Reference the JIRA ticket if appropriate

Description
-----------
If no JIRA ticket is referenced, describe the changes made. Note anything that you want the reviewers to know while
reviewing your pull request

After making a pull request
---------------------------
- [ ] If appropriate, put the link to the PR in the JIRA ticket
- [ ] Assign someone to review unless the change is trivial

orosz-usgs committed:
IOW-556  Added tables: field_visit_readings_by_loc and datum_converted_values

ssoper-usgs commented on liquibase/changeLogs/capture/tables/datumConvertedValues/datumConvertedValues.sql:
```diff
@@ -0,0 +1,12 @@
+create table ${AQTS_SCHEMA_NAME}.datum_converted_values
+(datum_converted_values_id                int8 NOT NULL GENERATED BY DEFAULT AS IDENTITY,
+json_data_id                              int8 NOT NULL,
```
Rather than `int8` on lines 2 and 3, I think the other tables use `bigint` instead.  It's unclear to me if `int8` is the same thing but the documentation seems to imply it?

https://www.postgresql.org/docs/11/datatype-numeric.html

Also, let's add a `(start with 1)` to line 2 after `IDENTITY` - see this other table for an example: https://github.com/usgs/aqts-capture-db/blob/master/liquibase/changeLogs/capture/tables/fieldVisitReadings/fieldVisitReadings.sql#L3

Also I think we can kill all the `NOT NULL` declarations in this file.

ssoper-usgs commented on liquibase/changeLogs/capture/tables/datumConvertedValues/datumConvertedValues.sql:
```diff
@@ -0,0 +1,12 @@
+field_visit_identifier                    text NOT NULL,
+target_datum                              text NOT NULL,
+unit                                      text NOT NULL,
+numeric_value                             numeric NOT NULL,
```
I don't think we want to carry forward the `numeric_value` in these etls, but rather just stick with the text type `display_value` below.  That way we have a lesser chance of accidentally mutating the AQTS number somewhere along the line, and we just rely on the `display_value` as the source of truth.

ssoper-usgs commented on liquibase/changeLogs/capture/tables/datumConvertedValues/datumConvertedValues.sql:
```diff
@@ -0,0 +1,12 @@
+partition_number                          integer NOT NULL,
+primary key (datum_converted_values_id, partition_number)
+)
+PARTITION BY LIST (partition_number);
```
Style fuss - in most of the other changesets we tend to just keep the postgres reserved words lowercase (line 12 and 2).  I don't have a strong opinion here, just pointing it out if we want to follow the established convention.

ssoper-usgs commented on liquibase/changeLogs/capture/tables/datumConvertedValues/datumConvertedValues.sql:
```diff
@@ -0,0 +1,12 @@
+create table ${AQTS_SCHEMA_NAME}.datum_converted_values
```
As a double insurance policy - I know you already have the precondition set in the changeset that creates this table, but let's `if not exists` to this create table statement as well.

ssoper-usgs commented on liquibase/changeLogs/capture/tables/fieldVisitReadingsByLoc/fieldVisitReadingsByLoc.sql:
```diff
@@ -0,0 +1,31 @@
+partition_number                          integer NOT NULL,
+primary key (field_visit_readings_by_loc_id, partition_number)
+)
+PARTITION BY LIST (partition_number);
```
See the feedback on the other table in this PR.

ssoper-usgs commented:
Let's add the partitioned tables to the pruning function here: https://github.com/usgs/aqts-capture-db/blob/master/liquibase/changeLogs/capture/functions/pruneTimeSeriesData/pruneTimeSeriesData.sql#L21

orosz-usgs committed:
IOW-556  Review changes
    Sql scripts: removed not null, added: if not exists & starts with
    Use lowercase and bigint instead of int8
    added new tables to pruner function.

ssoper-usgs commented on liquibase/changeLogs/capture/functions/pruneTimeSeriesData/pruneTimeSeriesData.sql:
```diff
@@ -32,6 +32,8 @@ begin
 	execute format('create table if not exists %I.field_visit_header_info_%s partition of %I.field_visit_header_info for values in (%L)', '${AQTS_SCHEMA_NAME}', month_name, '${AQTS_SCHEMA_NAME}', month_number);
 	execute format('create table if not exists %I.field_visit_readings_%s partition of %I.field_visit_readings for values in (%L)', '${AQTS_SCHEMA_NAME}', month_name, '${AQTS_SCHEMA_NAME}', month_number);
+	execute format('create table if not exists %I.field_visit_readings_%s partition of %I.field_visit_readings_by_loc for values in (%L)', '${AQTS_SCHEMA_NAME}', month_name, '${AQTS_SCHEMA_NAME}', month_number);
+	execute format('create table if not exists %I.field_visit_readings_%s partition of %I.datum_converted_values for values in (%L)', '${AQTS_SCHEMA_NAME}', month_name, '${AQTS_SCHEMA_NAME}', month_number);
```
I think here you'll want `field_visit_readings_by_loc` on line 35 rather than `field_visit_readings` and similarly on line 36 you'll want `datum_converted_values` rather than `field_visit_readings` just after the `if not exists` statements.

ssoper-usgs replied:
Further up just after line 21, you'll want to drop the `field_visit_readings_by_loc` and `datum_converted_values` tables as well.

orosz-usgs committed:
IOW-556  Added drop of new tables to  pruner function.

orosz-usgs committed:
IOW-556  Fixed typo on tables names in pruneTimeSeriesData.sql
